src/pages/index.js

```diff
@@ -1,127 +1,248 @@
+deleteCardPopup.setEventListeners();
+
+// Add Card  ===============================================================
+
+const addCardPopup = new PopupWithForm(selectors.addCardModal, (data) => {
+  setSubmitButtonText(addCardSubmitBtn, "Creating...");
+
+  api
+    .addCard(data)
+    .then((data) => {
+      const newCard = renderCard(data, userId);
+      sectionInstance.addItem(newCard);
+    })
+    .catch((err) => {
+      console.error(err.status);
+    })
+    .then(() => {
+      addCardPopup.close();
+      formValidators["add-card-form"].disableButton()
+    })
+    .finally(() => {
+      setSubmitButtonText(addCardSubmitBtn, "Create");
+    });
+});
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
+addCardBtn.addEventListener("click", () => {
+  addCardPopup.open();
+});
 
-      const card = renderCard(data)
+addCardPopup.setEventListeners();
 
-      cardSection.addItem(card);
-    },
-  },
+// Profile  ================================================================
 
-  selectors.cardList
+const userInfo = new UserInfo(
+  selectors.profileTitle,
+  selectors.profileDescription,
+  selectors.profileImage
 );
 
-cardSection.renderItems(initialCards);
-
-// Add Card Form ===============================================================
-
-
-const addCardFormPopup = new PopupWithForm(selectors.addCardModal, (data) => {
-  const newCard = renderCard(data)
+let sectionInstance;
+let userId;
+
+const editProfilePopup = new PopupWithForm(selectors.profileModal, (data) => {
+  setSubmitButtonText(profileModalSubmitBtn, "Saving...");
+  console.log(data);
+  api
+    .editProfile(data)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      editProfilePopup.close();
+      formValidators["profile-edit-form"].disableButton();
+    
+    })
+    .catch((err) => {
+      console.error(err.status);
+    })
+    .finally(() => {
+      setSubmitButtonText(profileModalSubmitBtn, "Save");
+    });
+});
 
-  cardSection.addItem(newCard);
+editProfilePopup.setEventListeners();
 
-  addCardFormPopup.close();
-  addFormValidator.toggleButtonState();
+editBtn.addEventListener("click", () => {
+  const profileData = userInfo.getUserInfo();
+  profileTitleInput.value = profileData.name;
+  profileDescriptionInput.value = profileData.info;
+  editProfilePopup.open();
 });
 
-addCardFormPopup.setEventListeners();
-
+// Card Section ================================================================
 
+api.loadData().then(([cards, userData]) => {
+  userId = userData._id;
+  userInfo.setUserInfo(userData.name, userData.about);
+  userInfo.setUserImage(userData.avatar);
+  sectionInstance = new Section(
+    {
+      items: cards,
+      renderer: (data) => {
+        const card = renderCard(data, userId);
+        sectionInstance.addItem(card);
+      },
+    },
+    selectors.cardList
+  );
+  sectionInstance.renderItems(cards);
+});
```

Comment:
* You need to `catch` possible errors at the very end of any server request in `index.js`